spec/presenters/summary/sections/contact_details_spec.rb

```diff
@@ -70,28 +74,62 @@
+        expect(answers[3].question).to eq(:telephone_number)
+        expect(answers[3].change_path).to match('applications/12345/steps/client/contact_details')
+        expect(answers[3].value).to eq('123456789')
+      end
+    end
 
-      expect(answers[0]).to be_an_instance_of(Summary::Components::FreeTextAnswer)
-      expect(answers[0].question).to eq(:home_address)
-      expect(answers[0].change_path).to match('applications/12345/steps/address/details/ff53a8dd')
-      expect(answers[0].value).to eq("Test\r\nHome\r\nCity\r\nPostcode\r\nCountry")
+    context 'when there is a residence type' do
+      let(:residence_type) { ResidenceType::PARENTS.to_s }
 
-      expect(answers[1]).to be_an_instance_of(Summary::Components::ValueAnswer)
-      expect(answers[1].question).to eq(:correspondence_address_type)
-      expect(answers[1].change_path).to match('applications/12345/steps/client/contact_details')
-      expect(answers[1].value).to eq('other_address')
+      it 'has the correct rows' do
+        expect(answers.count).to eq(5)
 
-      expect(answers[2]).to be_an_instance_of(Summary::Components::FreeTextAnswer)
-      expect(answers[2].question).to eq(:correspondence_address)
-      expect(answers[2].change_path).to match('applications/12345/steps/address/results/e77e4fa3')
-      expect(answers[2].value).to eq("Test\r\nCorrespondence\r\nCity\r\nPostcode\r\nCountry")
+        expect(answers[0]).to be_an_instance_of(Summary::Components::ValueAnswer)
+        expect(answers[0].question).to eq(:residence_type)
+        expect(answers[0].change_path).to match('/applications/12345/steps/client/residence_type')
+        expect(answers[0].value).to eq('parents')
+      end
 
-      expect(answers[3]).to be_an_instance_of(Summary::Components::FreeTextAnswer)
-      expect(answers[3].question).to eq(:telephone_number)
-      expect(answers[3].change_path).to match('applications/12345/steps/client/contact_details')
-      expect(answers[3].value).to eq('123456789')
+      context 'when residence type is `someone else`' do
+        let(:residence_type) { ResidenceType::SOMEONE_ELSE.to_s }
+        let(:relationship_to_someone_else) { 'A friend' }
+
+        it 'has the correct rows' do
+          expect(answers.count).to eq(6)
+
+          expect(answers[0]).to be_an_instance_of(Summary::Components::ValueAnswer)
+          expect(answers[0].question).to eq(:residence_type)
+          expect(answers[0].change_path).to match('/applications/12345/steps/client/residence_type')
+          expect(answers[0].value).to eq('someone_else')
+
+          expect(answers[1]).to be_an_instance_of(Summary::Components::FreeTextAnswer)
+          expect(answers[1].question).to eq(:relationship_to_someone_else)
+          expect(answers[1].change_path).to match('/applications/12345/steps/client/residence_type')
+          expect(answers[1].value).to eq('A friend')
+        end
+      end
```

Comment:
👇 Could you add ` let(:residence_type) { ResidenceType::NONE.to_s }` to the next context block for completion and to make it clear that that value will be set to be on that path?

Reply:
Or whatever order of HomeAddress / Residence status combos works for the new world with the new page AND the old world regression for apps without it

Reply:
Addressed! but this spec could do with restructuring maybe to make it clear when we're testing the view of historical / post residence type apps